### PR TITLE
Add Support for Multiple TimeSeries

### PR DESCRIPTION
**Pull request by Alrobbertz**
This PR relaxes the assumption that there can only be one `Epoch` variable within a HERMES data container object and HERMES CDF file. This is achieved through enabling the `timeseries` class member to be a `dict[string, astropy.TimeSeries]` where the key is the name of the intended epoch variable, and the value is the Time Series of measurements associated with that epoch variable. 

**Changes in HermesData**
- Update functions to handle `timeseries` as single item or dict as needed
- remove global meta from the main `timeseries` to create a unique `self._meta` structure to handle global metadata. 

**Changes in Schema**
- Updates `_get_depend()` to dynamically identify the correct Epoch for the given measurement

**Changes in IO**
- Update `CDFHandler` IO to load and save CDF files with multiple Epoch variables and time scales.

### CODE REVIEW

File: hermes_core/timedata.py [hunk 1]
```diff
@@ -107,48 +109,35 @@ def __init__(
...
-                raise TypeError(
-                    f"Column '{colname}' must be an astropy.units.Quantity object"
-                )
-            # Verify that the Column is only a single dimension
-            if len(timeseries[colname].shape) > 1:  # If there is more than 1 Dimension
-                raise ValueError(
-                    f"Column '{colname}' must be a one-dimensional measurement. Split additional dimensions into unique measurenents."
-                )
+        if isinstance(timeseries, dict):
+            for key, value in timeseries.items():
```
Comment (Alrobbertz) [inline hermes_core/timedata.py:113]: Validate that at least one TimeSeries dict has a primary "Epoch"

File: hermes_core/timedata.py [hunk 2]
```diff
@@ -532,13 +636,19 @@ def add_measurement(self, measure_name: str, data: u.Quantity, meta: dict = None
                 f"Column '{measure_name}' must be a one-dimensional measurement. Split additional dimensions into unique measurenents."
             )
 
-        self._timeseries[measure_name] = data
+        self._timeseries[hermes_core.DEFAULT_TIMESERIES_KEY][measure_name] = data
```
Comment (Alrobbertz) [inline hermes_core/timedata.py:639]: check which timeseries the measurement belongs in. also create an `add_timeseries`function.

**Comment by Alrobbertz** [comment]
Update Documentation and Add Coverage for new sections of code.

**Comment by Alrobbertz** [comment]
> Update Documentation and Add Coverage for new sections of code.

Updated!

**Review by ehsteve — APPROVED** [review]

File: hermes_core/timedata.py [hunk 3]
```diff
@@ -222,7 +212,23 @@ def __init__(
...
+        (`astropy.timeseries.TimeSeries` or `dict`) A `TimeSeries` representing one or more measurements as a function of time.
+        If there are multiple `TimeSeries`, a dictionary is returned.
+        """
+        if len(self._timeseries) > 1:
+            return {key: value for key, value in self._timeseries.items()}
+        else:
+            return self._timeseries[hermes_core.DEFAULT_TIMESERIES_KEY]
+
+    @property
+    def timeseries_dict(self):
```
[the remark this answers is not in the document]
Comment (Alrobbertz) [reply hermes_core/timedata.py]: Already pulled this out!

File: hermes_core/timedata.py [hunk 4]
```diff
@@ -256,14 +262,14 @@ def meta(self):
         """
         (`collections.OrderedDict`) Global metadata associated with the measurement data.
         """
-        return self._timeseries.meta
+        return self._meta
 
     @property
     def time(self):
```
[the remark this answers is not in the document]
Comment (Alrobbertz) [reply hermes_core/timedata.py:251]: Yes, this will just return the main Epoch time. Do you think it would be better to return a `dict[epoch, Time]`?

**Comment by Alrobbertz** [comment]
@kbromund and I were able to read his bin packet files into a `HermesData` object using this PR's code changes and it generated a CDF file containing the expected data. 

Ken and I agreed the process for updating variable metadata attributes is non-ideal with this data structure. It requires you to define the Epoch within the TimeSeries dictionary that contains the desired variable. For example, previously you could update variable attributes similar to the following:
`nemisis_data.timeseries["hermes_nem_fg_b1"].meta.update(...)`

However, if two Epochs are within the `.timeseries` member, you must now update variable attributes similar to:
`nemisis_data.timeseries["Epoch"]["hermes_nem_fg_b1"].meta.update(...}`

For now, I am going to add this nuance to the documentation, since this is more so a verbosity issue than a functionality issue. In the future, I'd like to work with Ken to have a function that can make this process smoother or less verbose and more dynamic. 

Once I update the documentation, I will merge this PR and create a new Issue to handle the dynamic Epoch identification for a given variable.

**Pull request merged**